app/src/main/java/de/danoeh/antennapod/discovery/PodcastIndexPodcastSearcher.java

```diff
@@ -0,0 +1,127 @@
+            String apiHeaderTime = String.valueOf(secondsSinceEpoch);
+            String data4Hash = API_KEY + API_SECRET + apiHeaderTime;
+            String hashString = sha1(data4Hash);
+
+            String encodedQuery;
+            try {
+                encodedQuery = URLEncoder.encode(query, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                // this won't ever be thrown
+                encodedQuery = query;
+            }
+
+            String formattedUrl = String.format(PODCASTINDEX_API_URL, encodedQuery);
+
+            OkHttpClient client = AntennapodHttpClient.getHttpClient();
+            Request.Builder httpReq = new Request.Builder()
+                    .addHeader("X-Auth-Date", apiHeaderTime)
+                    .addHeader("X-Auth-Key", API_KEY)
+                    .addHeader("Authorization", hashString)
+                    .addHeader("User-Agent", ClientConfig.USER_AGENT)
+                    .url(formattedUrl);
+            List<PodcastSearchResult> podcasts = new ArrayList<>();
+            try {
+                Response response = client.newCall(httpReq.build()).execute();
+
+                if (response.isSuccessful()) {
+                    String resultString = response.body().string();
+                    JSONObject result = new JSONObject(resultString);
+                    JSONArray j = result.getJSONArray("feeds");
+
+                    for (int i = 0; i < j.length(); i++) {
+                        JSONObject podcastJson = j.getJSONObject(i);
+                        PodcastSearchResult podcast = PodcastSearchResult.fromPodcastIndex(podcastJson);
+                        if (podcast.feedUrl != null) {
+                            podcasts.add(podcast);
+                        }
+                    }
+                } else {
+                    subscriber.onError(new IOException(response.toString()));
+                }
+            } catch (IOException | JSONException e) {
+                subscriber.onError(e);
+            }
+            subscriber.onSuccess(podcasts);
+        })
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread());
+    }
+
+    @Override
+    public Single<String> lookupUrl(String url) {
+        return Single.just(url);
+    }
+
+    @Override
+    public boolean urlNeedsLookup(String url) {
+        return false;
+    }
+
+    @Override
+    public String getName() {
+        return "Podcastindex.org";
+    }
+
+    public static String sha1(String clearString) {
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-1");
+            messageDigest.update(clearString.getBytes("UTF-8"));
+            return byteArrayToString(messageDigest.digest());
+        } catch (Exception ignored) {
+            ignored.printStackTrace();
+            return null;
+        }
+    }
+
+    public static String byteArrayToString(byte[] bytes) {
```

Comment:
This converts to a hex string, right? If so, please rename the function to be more clear. Just converting to string would be something like `new String(bytes)`

Reply:
Renamed it to `toHex`